extensions-core/namespace-lookup/src/main/java/io/druid/server/lookup/namespace/NamespacedExtractionModule.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.lookup.namespace;
+
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Binder;
+import com.google.inject.Key;
+import com.google.inject.TypeLiteral;
+import com.google.inject.multibindings.MapBinder;
+import io.druid.guice.LazySingleton;
+import io.druid.guice.PolyBind;
+import io.druid.initialization.DruidModule;
+import io.druid.query.lookup.NamespaceLookupExtractorFactory;
+import io.druid.query.lookup.namespace.ExtractionNamespace;
+import io.druid.query.lookup.namespace.ExtractionNamespaceCacheFactory;
+import io.druid.query.lookup.namespace.JDBCExtractionNamespace;
+import io.druid.query.lookup.namespace.URIExtractionNamespace;
+import io.druid.server.lookup.namespace.cache.NamespaceExtractionCacheManager;
+import io.druid.server.lookup.namespace.cache.OffHeapNamespaceExtractionCacheManager;
+import io.druid.server.lookup.namespace.cache.OnHeapNamespaceExtractionCacheManager;
+import java.util.List;
+
+/**
+ *
+ */
+public class NamespacedExtractionModule implements DruidModule
+{
+  private static final String TYPE_PREFIX = "druid.query.extraction.namespace.cache.type";
+
+  @Override
+  public List<? extends Module> getJacksonModules()
+  {
+    return ImmutableList.<Module>of(
+        new SimpleModule("DruidNamespacedExtractionModule")
+            .registerSubtypes(
+                ExtractionNamespace.class,
```

Comment:
it is not because it bother me, but if i am learning druid code and it happen to start reading this class i will be like what the heck the instruction means...so why having a confusing instruction. Hope you get my point.